### PR TITLE
rewrite home blueprint

### PR DESCRIPTION
**Pull request by tino097**
Clean home blueprint from #3762

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/tests/config/test_middleware.py [hunk 1]
```diff
@@ -391,11 +398,18 @@ def test_flask_core_and_pylons_core_route_is_served_by_flask(self):
...
+        environ = {
+            'PATH_INFO': '/about',
+            'REQUEST_METHOD': 'GET',
+        }
+        wsgiref.util.setup_testing_defaults(environ)
 
-        eq_(res.environ['ckan.app'], 'flask_app')
-        eq_(res.body, 'This was served from Flask')
+        answers = app.ask_around(environ)
+        print answers
```
Comment (amercader) [inline ckan/tests/config/test_middleware.py]: remove this statement

**Comment by amercader** [comment]
@tino097 Tests failures are caused by a separate i18n issue described in #3905. The rest of the PR looks fine, so I suggest parking it for now until we fix the `ITranslation` issue.

File: ckan/tests/config/test_middleware.py (hunk 1 above)
Comment (tino097) [reply ckan/tests/config/test_middleware.py]: done

**Comment by tino097** [comment]
@amercader it looks like extending of index.html is the problem for the last test failing, any suggestion ?

**Comment by amercader** [comment]
@tino097 you are right, `{% ckan_extends %}` is not being called if the template is served by Flask: https://github.com/ckan/ckan/issues/4044

**Comment by tino097** [comment]
@amercader all green!!!

**Comment by amercader** [comment]
@tino097 I'll prefer to keep the tests in test_middleware.py closer to their original intent. Can you add this change that doesn't change them so much?

```diff
diff --git a/ckan/tests/config/test_middleware.py b/ckan/tests/config/test_middleware.py
index 2dcead0..341d8d5 100644
--- a/ckan/tests/config/test_middleware.py
+++ b/ckan/tests/config/test_middleware.py
@@ -106,7 +106,7 @@ class TestAppDispatcher(helpers.FunctionalTestBase):
             return 'This was served from Flask'
 
         # This endpoint is defined both in Flask and in Pylons core
-        flask_app.add_url_rule('/about', view_func=test_view)
+        flask_app.add_url_rule('/flask_core', view_func=test_view)
 
         # This endpoint is defined both in Flask and a Pylons extension
         flask_app.add_url_rule('/pylons_and_flask', view_func=test_view)
@@ -114,16 +114,9 @@ class TestAppDispatcher(helpers.FunctionalTestBase):
     def test_ask_around_is_called(self):
 
         app = self._get_test_app()
-        ckan_app = app.app
-        start_response = mock.MagicMock()
-        environ = {
-            'PATH_INFO': '/',
-            'REQUEST_METHOD': 'GET',
-        }
-        wsgiref.util.setup_testing_defaults(environ)
         with mock.patch.object(AskAppDispatcherMiddleware, 'ask_around') as \
                 mock_ask_around:
-            ckan_app(environ, start_response)
+            app.get('/', status=404)
 
             assert mock_ask_around.called
 
@@ -398,17 +391,11 @@ class TestAppDispatcher(helpers.FunctionalTestBase):
         This should never happen in core, but just in case
         '''
         app = self._get_test_app()
-        app = app.app
 
-        environ = {
-            'PATH_INFO': '/about',
-            'REQUEST_METHOD': 'GET',
-        }
-        wsgiref.util.setup_testing_defaults(environ)
+        res = app.get('/flask_core')
 
-        answers = app.ask_around(environ)
-
-        eq_(answers, [(True, 'flask_app', 'core'), (True, 'pylons_app', 'core')])
+        eq_(res.environ['ckan.app'], 'flask_app')
+        eq_(res.body, 'This was served from Flask')
 
 
 class TestFlaskUserIdentifiedInRequest(helpers.FunctionalTestBase):

```

Then we can merge!

**Comment by tino097** [comment]
Next song: Sugarbabes - Push the Button :)

**Pull request merged**